### ISSUE TITLE
Compose post button is missing accessibilityLabel on profile views (mobile app)

### ISSUE THREAD

**Issue by hs4man21** [bluesky-social/social-app#695]
**Describe the bug**
If I'm in the feed tab, then the compose post button has the appropriate `accessibilityLabel` for screen reader users.

However, the label is missing when I am trying to post while viewing someone's profile, including my own (see video below).

**To Reproduce**

1. Use an iPhone (or Android with TalkBack screen reader turned on, but my instructions assume iPhone).
2. Turn on [the VoiceOver screen reader](https://build.washingtonpost.com/resources/accessibility/screen-readers#iOS) in your iPhone's accessibility settings.
3. Open the Bluesky app.
4. Tap the compose button to hear its label read. Now go to your profile page or someone else's profile via search, feed, etc. and click the compose button. You should notice that it now has no label, so it's unknown to screen reader users what the button does.

**Expected behavior**

The `accessibilityLabel` should exist and be the same anywhere that the compose post button appears.

**Screenshots**

https://github.com/bluesky-social/social-app/assets/38192823/e6051a22-1291-43ea-a5af-46cb9070ea9c

**Details**

- Platform: iPhone XR, iOS 16.4.1, using VoiceOver screen reader
- Platform version: MacOS: Ventura 13.1, Chrome: 112.0.5615.137
- App version: 1.28
